packages/cli/src/models/status/StatusChecker.js

```diff
@@ -106,23 +114,43 @@ export default class StatusChecker {
     this._checkUnregisteredLocalProxies(proxiesInfo)
   }
 
-  async _checkRemoteImplementation({ alias, address }) {
+  _checkRemoteContractImplementation({ alias, address }, bytecode) {
     if (this.networkFile.hasContract(alias)) {
-      this._checkImplementationAddress(alias, address)
-      await this._checkImplementationBytecode(alias, address)
+      this._checkContractImplementationAddress(alias, address)
+      this._checkContractImplementationBytecode(alias, address, bytecode)
     }
-    else this.visitor.onMissingRemoteContract('none', 'one', { alias, address })
+    else this.visitor.onMissingRemoteImplementation('none', 'one', { alias, address })
   }
 
-  _checkImplementationAddress(alias, address) {
+  _checkContractImplementationAddress(alias, address) {
     const expected = this.networkFile.contract(alias).address
-    if (address !== expected) this.visitor.onMismatchingContractAddress(expected, address, { alias, address })
+    if (address !== expected) this.visitor.onMismatchingImplementationAddress(expected, address, { alias, address })
   }
 
-  async _checkImplementationBytecode(alias, address) {
+  _checkContractImplementationBytecode(alias, address, bytecode) {
     const expected = this.networkFile.contract(alias).bodyBytecodeHash
-    const observed = bytecodeDigest(await promisify(web3.eth.getCode.bind(web3.eth))(address))
-    if (observed !== expected) this.visitor.onMismatchingContractBodyBytecode(expected, observed, { alias, address, bodyBytecodeHash: observed })
+    const observed = bytecodeDigest(bytecode)
+    if (observed !== expected) this.visitor.onMismatchingImplementationBodyBytecode(expected, observed, { alias, address, bodyBytecodeHash: observed })
+  }
+
+  _checkRemoteSolidityLibImplementation({ alias, address }, bytecode) {
+    if (this.networkFile.hasSolidityLib(alias)) {
+      this._checkSolidityLibImplementationAddress(alias, address)
+      this._checkSolidityLibImplementationBytecode(alias, address, bytecode)
+    }
+    //TODO: implement missing remote solidity libs validation
```

Comment:
can we move this to an issue?

Reply:
https://github.com/zeppelinos/zos/issues/244